### PR TITLE
Few File component drag and drop

### PR DESCRIPTION
**Pull request by dawoodkhan82**
## Description
- Fixes Drag and Drop for types of the first element in file_types
- Added playwright playwright tests for drag & drop

Closes: #7094

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: js/file/shared/File.svelte [hunk 1]
```diff
@@ -20,7 +20,7 @@
 	label={label || "File"}
 />
 
-{#if value && (Array.isArray(value) ? value.length > 0 : true)}
+{#if (Array.isArray(value) && value.length > 0) || value !== null}
```
Comment (abidlabs) [inline js/file/shared/File.svelte:23]: #7132 has already been fixed so I don't think we need to make this change?

**Comment by abidlabs** [comment]
@dawoodkhan82 #7132 is already fixed, so I've removed it from the parent comment, and I think we can reduce some of the changes in this PR as a result

File: js/file/shared/File.svelte (hunk 1 above)
Comment (dawoodkhan82) [reply js/file/shared/File.svelte:23]: Either is fine, but I can remove my change.

Comment (abidlabs) [reply js/file/shared/File.svelte]: The original seems slightly simpler to parse and we've already validated it, so maybe keep the original? Not a big deal either way

**Comment by abidlabs** [comment]
Thanks @dawoodkhan82. I tested this and can confirm that the original issue is fixed. However, the drag-and-drop doesn't seem to work with file extensions. Here's the repro:

```py
import gradio as gr
def generate_file(file_obj):
    return None

inputs = gr.components.File( file_types = [ '.jpg', '.mov' ] )
outputs = gr.components.File()
demo = gr.Interface(fn=generate_file, inputs=inputs, outputs=outputs)
demo.launch()
```

If you try dragging-and-dropping `.jpg` or `.mov` files, they won't get uploaded (however, if you click on the file component, they'll still work)

**Comment by dawoodkhan82** [comment]
@abidlabs Should be fixed

**Comment by abidlabs** [comment]
Extensions work but hmm now the original repro isn't working for me:

```py
import gradio as gr
def generate_file(file_obj):
    return None

inputs = gr.components.File( file_types = [ 'image', 'video' ] )
outputs = gr.components.File()
demo = gr.Interface(fn=generate_file, inputs=inputs, outputs=outputs)
demo.launch()
```

**Comment by dawoodkhan82** [comment]
@abidlabs Ah yes I see what's going wrong. Need to pass both the file extension and file type of the uploaded file. Pushing a fix now.

**Comment by dawoodkhan82** [comment]
@abidlabs Can you double check and make sure it works for you now?

**Review by abidlabs — APPROVED** [review]
Nice! Works for me now @dawoodkhan82 thanks

**Pull request merged**